frontend/controller/actions/group.js

```diff
@@ -53,25 +61,29 @@ export default sbp('sbp/selectors/register', {
           groupName: name,
           groupPicture: finalPicture,
           sharedValues,
-          mincomeAmount: +mincomeAmount, // ensure this is a number
+          mincomeAmount: +mincomeAmount,
           mincomeCurrency: mincomeCurrency,
           proposals: {
-            // TODO: make the UI support changing the rule type, so that we have
-            //       a component for RULE_DISAGREEMENT as well
-            [PROPOSAL_GROUP_SETTING_CHANGE]: merge({},
-              proposals[PROPOSAL_GROUP_SETTING_CHANGE].defaults,
-              { ruleSettings: { [RULE_PERCENTAGE]: { threshold: thresholdChange } } }
+            [PROPOSAL_GROUP_SETTING_CHANGE]: merge(
+              merge({}, proposals[PROPOSAL_GROUP_SETTING_CHANGE].defaults),
+              proposalSettings
+            ),
+            [PROPOSAL_INVITE_MEMBER]: merge(
+              merge({}, proposals[PROPOSAL_INVITE_MEMBER].defaults),
+              proposalSettings
             ),
-            [PROPOSAL_INVITE_MEMBER]: merge({},
-              proposals[PROPOSAL_INVITE_MEMBER].defaults,
-              { ruleSettings: { [RULE_PERCENTAGE]: { threshold: thresholdMemberApproval } } }
+            [PROPOSAL_REMOVE_MEMBER]: merge(
+              merge({}, proposals[PROPOSAL_REMOVE_MEMBER].defaults),
+              proposalSettings
```

Comment:
Note: `merge()` wasn't working as expected. It only accepts 2 parameters. So, we need to use `merge` twice: first from default settings to an empty object and then from (custom) settings to defaults settings.